cli/batch-score-rest.sh

```diff
@@ -0,0 +1,311 @@
+        \"resources\": {
+            \"instanceCount\": 1
+        },
+        \"maxConcurrencyPerInstance\": \"4\",
+        \"retrySettings\": {
+            \"maxRetries\": 3,
+            \"timeout\": \"PT30S\"
+        },
+        \"errorThreshold\": \"10\",
+        \"loggingLevel\": \"info\",
+        \"miniBatchSize\": \"5\",
+    }
+}")
+#</create_deployment>
+# TEMPORARY! AzureAsyncOperationUri is not being returned on Deployment
+sleep 60s
+#operation_id=$(echo $response | jq '.properties.properties.AzureAsyncOperationUri')
+#wait_for_completion $operation_id
+
+#<set_endpoint_defaults>
+response=$(curl --location --request PUT "https://management.azure.com/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE/batchEndpoints/$ENDPOINT_NAME?api-version=$API_VERSION" \
+--header "Content-Type: application/json" \
+--header "Authorization: Bearer $TOKEN" \
+--data-raw "{
+    \"properties\": {
+        \"authMode\": \"aadToken\",
+        \"defaults\": {
+            \"deploymentName\": \"$DEPLOYMENT_NAME\"
+        }
+    },
+    \"location\": \"$LOCATION\"
+}")
+
+operation_id=$(echo $response | jq -r '.properties' | jq -r '.properties' | jq -r '.AzureAsyncOperationUri')
+wait_for_completion $operation_id
+
+#</set_endpoint_defaults>
+
+# <get_endpoint>
+response=$(curl --location --request GET "https://management.azure.com/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE/batchEndpoints/$ENDPOINT_NAME?api-version=$API_VERSION" \
+--header "Content-Type: application/json" \
+--header "Authorization: Bearer $TOKEN")
+
+SCORING_URI=$(echo $response | jq -r '.properties.scoringUri')
+# </get_endpoint>
+
+# <get_access_token>
+SCORING_TOKEN=$(az account get-access-token --resource https://ml.azure.com --query accessToken -o tsv)
+# </get_access_token>
+
+# <score_endpoint_with_data_in_cloud>
+response=$(curl --location --request POST $SCORING_URI \
+--header "Authorization: Bearer $SCORING_TOKEN" \
+--header "Content-Type: application/json" \
+--data-raw "{
+    \"properties\": {
+        \"dataset\": {
+            \"dataInputType\": \"DataUrl\",
+            \"Path\": \"https://pipelinedata.blob.core.windows.net/sampledata/mnist\"
+        }
+    }
+}")
+
+JOB_ID=$(echo $response | jq -r '.id')
+JOB_ID_SUFFIX=$(echo ${JOB_ID##/*/})
+
+# </score_endpoint_with_data_in_cloud>
+# </score_endpoint>
+
+# <check_job>
+wait_for_job_completion $SCORING_URI/$JOB_ID_SUFFIX
+# </check_job>
+
+# <score_endpoint_with_dataset>
+response=$(curl --location --request POST $SCORING_URI \
+--header "Authorization: Bearer $TOKEN" \
+--header "Content-Type: application/json" \
+--data-raw "{
+    \"properties\": {
+        \"dataset\": {
+            \"dataInputType\": \"DatasetVersion\",
+            \"datasetName\": \"$DATASET_NAME\",
```

Comment:
need to create a dataset first before invoke

Reply:
Added dataset creation